cfme/tests/cloud_infra_common/test_power_control_rest.py

```diff
@@ -85,19 +93,20 @@ def test_stop(rest_api, vm_obj, from_detail):
     Metadata:
         test_flag: rest
     """
-    vm = vm_obj.get_vm_via_rest()
-    verify_vm_power_state(vm_obj, state=vm_obj.STATE_ON, action=vm.action.start)
+    vm_obj.wait_for_vm_state_change(desired_state=vm_obj.STATE_ON)
+    vm = rest_api.collections.vms.get(name=vm_obj.name)
 
     if from_detail:
         vm.action.stop()
     else:
-        vm_obj.get_collection_via_rest().action.stop(vm)
+        rest_api.collections.vms.action.stop(vm)
     verify_action_result(rest_api)
-    verify_vm_power_state(vm_obj, state=vm_obj.STATE_OFF)
+    wait_for_vm_state_change(vm_obj, vm_obj.STATE_OFF)
+    soft_assert(not verify_vm_power_state(vm, vm_obj.STATE_ON), "vm still running")
```

Comment:
OPTIONAL: I don't think you need soft_assert here since this is only existing check in this test